### PR TITLE
App manager improvements

### PR DESCRIPTION
**Pull request by PhilippLgh**
#### What does it do?
- fixes a bug where apps could not be started after main window is closed
- allows apps to define plugin dependencies and settings so that client are started with the correct config automatically

### CODE REVIEW

File: grid_apps/AppManager.js [hunk 1]
```diff
@@ -134,7 +183,7 @@ class AppManager extends EventEmitter {
 
     let url = app.url || 'http://localhost:3000'
     const mainWindow = createRenderer(
-      WindowManager.getMainUrl(),
+      await getGridUiUrl(), // FIXME might be very inefficient. might load many grid-ui packages into memory!!
```
Comment (marcgarreau) [inline grid_apps/AppManager.js]: Any ideas for this one?

Comment (marcgarreau) [reply grid_apps/AppManager.js]: After discussion at standup: okay to punt this to next release. We expect few windows to be created, so not critical at this stage.

**Comment by marcgarreau** [comment]
@evertonfraga you're right, that produced an error which was not surfaced to the user. I think a great solution will take some research, but in the meantime, I've at least given the user some feedback:

I'm okay shipping this in the upcoming release and revisiting more optimal user experiences in a subsequent release.

File: ethereum_clients/Plugin.js [hunk 2]
```diff
@@ -182,7 +183,37 @@ class Plugin extends EventEmitter {
...
+          }].\n\nPress 'OK' to allow this time.
+        `
+        },
+        async response => {
+          const userPermission = response !== 1 // = index of 'cancel'
+          if (userPermission) {
+            await this.start(flags, release)
+            resolve()
+          } else {
+            reject(new Error('user permission denied'))
```
Comment (evertonfraga) [inline ethereum_clients/Plugin.js]: strictly speaking, this should not generate an error message, as it's not an error. we can handle it better another time.

image

**Pull request merged**